pkg/settings/registry.go

```diff
@@ -517,11 +517,15 @@ var ReadableTypes = map[string]string{
 }
 
 // RedactedValue returns:
-//   - a string representation of the value, if the setting is reportable (or it
-//     is a string setting with an empty value);
-//   - "<redacted>" if the setting is not reportable;
+//   - a string representation of the value, if the setting is reportable;
+//   - "<redacted>" if the setting is not reportable, sensitive, or a string;
 //   - "<unknown>" if there is no setting with this name.
 func RedactedValue(key InternalKey, values *Values, forSystemTenant bool) string {
+	if k, ok := registry[key]; ok {
+		if k.Typ() == "s" || k.isSensitive() || !k.isReportable() {
```

Comment:
Longer term, are we going to continue using masking, sensitive properties of settings?

Will they only be used when trying to `SHOW CLUSTER SETTING`?

Reply:
For diagnostics, we will rely on `RedactedValue` for enforcement and the fact that `MaskedSetting` is used here is an implementation detail.

`SHOW CLUSTER SETTING` relies on `MaskedSetting` for enforcement.